src/tests/ipc_tests.rs

```diff
@@ -408,4 +411,213 @@ pub mod ipc_tests {
         assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
         lindrustfinalize();
     }
+
+    #[test]
+    pub fn ut_lind_ipc_pipe2_nonblock() {
```

Comment:
there should be at least one test which does a bunch of common error conditions.  And also a test with a unis domain socket...

Reply:
this is testing for EAGAIN which is really the only error condition besides EPIPE which we can't test on standalone RustPOSIX because it will trigger a signal.

Reply:
There's also a few Unix tests already, though David can flesh them out further when he gets to send/recv.

Reply:
I guess technically the pipe() syscall can error with ENFILE but thats really more a `get_next_fd()` test than a pipe() test.